nevergrad/optimization/recaster.py

```diff
@@ -54,9 +54,12 @@ def run(self) -> None:
         try:
             self.output = self._caller(self._fake_callable, *self._args, **self._kwargs)
         except StopOptimizerThread:  # gracefully stopping the thread
-            pass
+            self.messages_ask.put(ValueError("Optimization told to finish"))
         except Exception as e:  # pylint: disable=broad-except
+            self.messages_ask.put(e)
             self.error = e
+        else:
+            self.messages_ask.put(ValueError("Optimization finished"))
```

Comment:
can you explain this? I don't understand the aim here
Wouldnt that interfere with retries (aka starting the optimization again from a different starting point if it finished ahead of budget)?

Reply:
But you're right, that in this case we really want to go into the is_alive=False branch rather than pass an error to the user. I'll use a special value.